### PR TITLE
JFG-889: JaaS Environment API

### PR DESCRIPTION
**Pull request by anton-antonenko**
Refactoring of Dao;
Add TestEnvironment model, dao and service;
Add tests for Dao;
Disable some PMD checks.

### CODE REVIEW

File: chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/JHttpQuery.java [hunk 1]
```diff
@@ -212,6 +214,23 @@
...
+     *
+     * @param key   name of header
+     * @param value value of header
+     * @return this
+     * @apiNote Usage:
+     * <pre>{@code
+     * JHttpQuery httpQuery = new JHttpQuery().header("header", "value");
+     * }</pre>
+     */
+    public JHttpQuery<T> header(String key, String value) {
```
Comment (dlatnikov) [inline chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/JHttpQuery.java]: why is it here? it was a part of other story

File: jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.web.servlet.support.ServletUriComponentsBuilder.fromCurrentRequest;
+
+@RestController
+@RequestMapping(value = "/envs")
+public class TestEnvironmentRestController extends AbstractController {
+
+    //private static final Logger LOGGER = LoggerFactory.getLogger(TestEnvironmentRestController.class);
```
Comment (dlatnikov) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java]: do you need it here

File: jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        TestEnvironmentEntity created = testEnvService.create(testEnv);
+        setExpiresHeader(response, created);
+        setSessionCookie(response, created, environmentsTtlMinutes * 60);
+        return ResponseEntity.created(fromCurrentRequest().path("/{envId}")
+                .buildAndExpand(testEnv.getEnvironmentId()).toUri()).build();
+    }
+
+    private void validateTestEnvId(TestEnvironmentEntity testEnv) {
+        String envId = testEnv.getEnvironmentId();
+        Pattern envIdPattern = Pattern.compile("^[a-zA-Z0-9\\._\\-]{1,249}$");
```
Comment (a-badaev) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java]: can be cached

File: jaas/src/main/java/com/griddynamics/jagger/jaas/exceptions/ResourceAlreadyExistsException.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.griddynamics.jagger.jaas.exceptions;
+
+import static java.lang.String.format;
+
+public class ResourceAlreadyExistsException extends RuntimeException {
+
+    public ResourceAlreadyExistsException(String resourceName, String resourceId) {
+        super(format("%s resource with id \'%s\' already exists.", resourceName, resourceId));
```
Comment (anton-antonenko) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/exceptions/ResourceAlreadyExistsException.java]: ok

File: jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java (hunk 3 above)
Comment (anton-antonenko) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java]: ok

**Comment by SokolAndrey** [comment]
🇶🇦

**Pull request merged**